roachtest: restart node in kv/quiesence unconditionally

tbg opened the pull request:
When the test fails, restart the node anyway, so we don't get distracted
by a faux "dead node".

Release note: None

tbg committed:
roachtest: restart node in kv/quiesence unconditionally

When the test fails, restart the node anyway, so we don't get distracted
by a faux "dead node".

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55239)
<!-- Reviewable:end -->

tbg committed:
roachtest: remove erroneous skip

roachtests are never invoked under race.

Release note: None

tbg committed:
roachtest: skip kv/gracefuldraining

It's reliably flaky.

Touches #53811.
Touches #54088.
Touches #53760.
Touches #53725.

Release note: None

tbg committed:
roachtest: de-flake (again again!) roachtest/inconsistent

The consistency check could still happen at a moment at which
not all nodes were live "to each other". A consistency check
that skips a node is between two nodes, so a 50:50 chance of
killing either. Not what we want - we want n1 to be the designated
loser.

Fixes #55205.

Release note: None

tbg commented:
Friendly ping.

knz reviewed (approved):
> Reviewed 1 of 1 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55239)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg commented:
bors r=knz
TFTR!
